File: 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/utils/timer/TickTimer.java
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.utils.timer;

public final class TickTimer {

    private int tick;

    public void update() {
        tick++;
    }

    public void reset() {
        tick = 0;
    }

    public boolean hasTimePassed(final int ticks) {
        return tick >= ticks;
    }
}

File: src/main/java/net/ccbluex/liquidbounce/utils/timer/TickTimer.kt
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.utils.timer

class TickTimer
{
    var tick = 0
        private set

    fun update()
    {
        tick++
    }

    fun reset()
    {
        tick = 0
    }

    fun hasTimePassed(ticks: Int): Boolean = tick >= ticks
}
